include/libtorrent/time.hpp

```diff
@@ -87,6 +87,33 @@ namespace libtorrent {
 	std::int64_t total_microseconds(T td)
 	{ return duration_cast<microseconds>(td).count(); }
 
+	template <typename HighResTimePoint>
```

Comment:
IIRC, these functions are clock type independent. So I would drop any mentions of "high resolution", e.g. `template <typename TimePoint>`, `unix_time_to_time_point` (or even `from_time_t` to be consistent with `to_time_t`).